src/mongocxx/private/index_view.hh

```diff
@@ -134,6 +133,25 @@ class index_view::impl {
             opts_doc.append(bsoncxx::builder::concatenate_doc{session->_get_impl().to_document()});
         }
 
+        if (options.commit_quorum()) {
+            auto server_description = libmongoc::client_select_server(
+                _client, true /* for_writes */, nullptr /* read_prefs */, &error);
+            auto is_master = libmongoc::server_description_ismaster(server_description);
+
+            bson_iter_t iter;
+            bson_iter_init_find(&iter, is_master, "maxWireVersion");
+            int32_t max_wire_version = bson_iter_int32(&iter);
+
+            if (max_wire_version < 9) {
+                throw logic_error{
```

Comment:
The server description is leaked if this throws, this can be destroyed by calling the C function directly:

```
mongoc_server_description_destroy(server_description);
```

Or, I think the more idiomatic C++ solution would be wrap the C type in something that calls `mongoc_server_description_destroy` in the destructor (similar to how `scoped_bson_t` wraps a `bson_t`). If it seems reasonable, you could add a private helper class in this file for that:

```
class scoped_server_description {
public:
   scoped_server_description (mongoc_server_description_t *sd) : sd (sd) {}
   ~scoped_server_description () { mongoc_server_description_destroy (sd); }
   mongoc_server_description_t* sd;
};
```

Then, changing the assignment above should prevent any leaks on early returns or throws:

```
scoped_server_description scoped_sd(
    libmongoc::client_select_server(_client, true /* for_writes */, nullptr /* read_prefs */, &error)
);
```

Reply:
Thanks for the code snippet. Done!